jsonpb/jsonpb.go

```diff
@@ -878,7 +878,9 @@ func (u *Unmarshaler) unmarshalValue(target reflect.Value, inputValue json.RawMe
 			target.Set(reflect.New(targetType.Elem()))
 			target = target.Elem()
 		}
-		target.SetInt(int64(n))
+		if targetType.Kind() == reflect.Int32 {
+			target.SetInt(int64(n))
+		}
```

Comment:
The conditional is correct, but the behavior is still off. This simply ignores the invalid input instead of returning an error.